pallet/src/lib.rs

```diff
@@ -158,11 +160,43 @@ pub mod pallet {
             <Authorities<T>>::put(authorities);
         }
 
-        pub fn next_session_authorities() -> Result<Vec<T::AuthorityId>, AlephApiError> {
+        pub fn next_session_authorities() -> Vec<T::AuthorityId> {
             Session::<T>::queued_keys()
                 .iter()
-                .map(|(_, key)| key.get(T::AuthorityId::ID).ok_or(AlephApiError::DecodeKey))
-                .collect::<Result<Vec<T::AuthorityId>, AlephApiError>>()
+                .map(|(_, key)| {
+                    key.get(T::AuthorityId::ID)
+                        .expect("Keys in storage decode correctly.")
+                })
+                .collect()
+        }
+
+        pub fn future_session_validator_ids(
+            session_id: u32,
+        ) -> Result<Vec<T::ValidatorId>, ApiError> {
+            let current_session = Session::<T>::current_index();
+            if session_id < current_session {
+                return Err(ApiError::TooLowSessionId);
+            }
+            // If the session_id is higher than the one for which the validator rotation is scheduled, we output
+            // the new validator set, otherwise we output the current set (we read it off the session pallet).
```

Comment:
It would be nice to document the behaviour, especially in context of potentially getting inaccurate information – it should be made clear to users that caching a response from this function about a session more than 2(?) away from the current session is a bad idea, since an authority change might happen.

Bonus question about the far future – will we at some point keep a more complex schedule of authorities? E.g. information about authorities for the next 50 sessions, with explicit lack of knowledge of later sessions? I'm asking because it would be nice to be able to return a `TooHighSessionId` error in such cases, rather than potentially providing inaccurate information (it's probably not a good idea now though).

Reply:
Replaced.